server/segment_rpc.go

```diff
@@ -506,13 +508,32 @@ func genSegCreds(sess *BroadcastSession, seg *stream.HLSSegment) (string, error)
 
 	// Generate serialized segment info
 	segData := &net.SegData{
-		ManifestId:   []byte(md.ManifestID),
-		Seq:          md.Seq,
-		Hash:         hash,
-		FullProfiles: fullProfiles,
-		Sig:          sig,
-		Storage:      storage,
+		ManifestId: []byte(md.ManifestID),
+		Seq:        md.Seq,
+		Hash:       hash,
+		Sig:        sig,
+		Storage:    storage,
+		// Triggers failure on Os that don't know how to use FullProfiles/2
+		Profiles: []byte("invalid"),
+	}
+	// If all outputs are mpegts, use the older SegData.FullProfiles field
+	// for compatibility with older orchestrators
+	allTS := true
+	for _, p := range sess.Profiles {
+		switch p.Format {
+		case ffmpeg.FormatNone: // default output is mpegts for FormatNone
+		case ffmpeg.FormatMPEGTS:
+		default:
+			allTS = false
+			break
```

Comment:
I believe this will only break out of the switch, but will not break out of the for loop since break statements only terminate the innermost for, switch, select (see https://golang.org/ref/spec#Break_statements). We'd have to label the loop to break out of the loop - something like:

```golang

allTSLoop:
for _, p := range sess.Profiles {
    switch p.Format {
        ...
        default:
            allTS = false
            break allTSLoop
    }
}

Reply:
Ah yeah. Am I the only one who thinks that switch-break is a nonsense feature for a language that doesn't have default switch case fallthrough? (Unlike, say, C/C++ which clearly inspired this behavior.) Fixed in fa819dfee0d05f25b5eee8d6734e42c2848b629f

Reply:
( further fixed up by 6358ceb403198e59cfd017fc8f48cd28c4de4f21  ; original changes didn't even compile )